template/pyproject.toml.jinja

```diff
@@ -241,6 +234,7 @@ extra-dependencies = [
     "mkdocs-awesome-pages-plugin ~=2.9",
 ]
 {%- elif documentation == "sphinx" %}
+python = "3.10"
```

Comment:
This can also be a list, e.g. `["3.10", "3.11", "3.12"]`, but I don't think we want to matrix docs builds. And hatch doesn't seem to support open ended version ranges `">=3.10"` :(

Reply:
I think eventually we'll want to start taking all these consts out of the templates themselves and into .yaml files that we load into the template contexts, so this is totally fine to me and in the future we can give ourselves a handle if we want to have better control over the python version in the generated project.